src/ReflectionParameter.php

```diff
@@ -338,50 +349,66 @@ public function isCallable()
     /**
      * @inheritDoc
      */
-    public function isDefaultValueAvailable()
+    public function isDefaultValueAvailable(): bool
     {
-        return isset($this->parameterNode->default);
+        if (! isset($this->parameterNode->default)) {
+            return false;
+        }
+
+        // start from PHP 8.1, isDefaultValueAvailable() returns false if next parameter is required
+        // see https://github.com/php/php-src/issues/8090
+        $parameters = $this->declaringFunction->getNode()->getParams();
+        foreach ($parameters as $key => $parameter) {
+            if ($key <= $this->parameterIndex) {
+                continue;
+            }
+
+            if (! $parameter->default instanceof Expr) {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
@lisachenko this is needed, as since PHP 8.1, `isDefaultValueAvailable()` returns false if next parameter is required, see 

- https://github.com/php/php-src/issues/8090

Reply:
Aha, I see... Thanks!